src/CacheUtility.js

```diff
@@ -0,0 +1,254 @@
+class CacheUtility {
+	static isAvailable(type) {
+		// Helper method to determine if both the store exists and has space
+		const storage = window[type];
+		try {
+			const x = '__storage_test__';
+			storage.setItem(x, x);
+			storage.removeItem(x);
+			return true;
+		} catch (e) {
+			return e instanceof DOMException && (
+				// everything except Firefox
+				e.code === 22 ||
+				// Firefox
+				e.code === 1014 ||
+				// test name field too, because code might not be present
+				// everything except Firefox
+				e.name === 'QuotaExceededError' ||
+				// Firefox
+				e.name === 'NS_ERROR_DOM_QUOTA_REACHED') &&
+				// acknowledge QuotaExceededError only if there's something already stored
+				storage.length !== 0;
+		}
+	}
+
+	constructor(config) {
+		// Determine the target storage entity on the window
+		this.target = window[config.target];
+
+		// Store the desired tiers from longest to shortest expiration
+		this.sortedTiers = config.tiers
+			.filter((tier) => {
+				// We don't want any tiers with white space in the name
+				if (tier.name.match(/\s/g)) {
+					console.error(`Invalid tier name "${tier.name}" with whitespace`);
+					return false;
+				}
+				return true;
+			})
+			.sort((a, b) => {
+				if (a.expiration < b.expiration) {
+					return 1;
+				}
+				if (a.expiration > b.expiration) {
+					return -1;
+				}
+				return 0;
+			});
+		/* Now that the tiers are sorted in decreasing expiration, we can
+		generate the map of tier name to key. This key is a concatendated list
+		of all of the tier names that precede the current tier. This allows us
+		to filter all tiers that should expire given a particular amount of inactivity */
+		this.tierMap = {};
+		this.sortedTiers.forEach((tier, index) => {
+			if (index === 0) {
+				this.tierMap[tier.name] = {
+					key: tier.name,
+					expiration: tier.expiration,
+				};
+			} else {
+				this.tierMap[tier.name] = {
+					key: `${this.tierMap[this.sortedTiers[index - 1].name].key}-${tier.name}`,
+					expiration: tier.expiration,
+				};
+			}
+		});
+
+		if (this.isAvailable()) {
```

Comment:
Technically, that is valid, too. I guess in my mind, I only wanted to have the static method used if the end user is trying to decide whether or not they even want to create an instance of CacheUtility. I wanted to have the isAvailable method tied to the instance of CacheUtility so I could use the instance to determine whether the cache is available rather than import CacheUtility just to have access to this method.

Reply:
I would not be opposed to rewriting the instance method to use the static method to reduce on code duplication if that helps?